src/core/lib/gprpp/arena.cc

```diff
@@ -31,19 +31,36 @@
 #include "src/core/lib/gpr/alloc.h"
 #include "src/core/lib/gprpp/memory.h"
 
+namespace grpc_core {
+
 template <size_t alignment>
-static void* gpr_arena_malloc(size_t size) {
+static void* _aligned_alloc(size_t size) {
+#if defined(GPR_LINUX)
```

Comment:
Instead of having this compile-time switch here, how about doing it in the definition of `gpr_malloc_aligned()` and `gpr_free_aligned()`?  That way, any code that calls those functions can get this benefit.

Reply:
Right now the only callers are the grpc_call arena, and the global (initialization time allocated) handshaker list. No other users at present.

My preference is to keep it here since:
1) I already changed gpr_malloc_aligned/gpr_free_aligned to use this also anyways,
2) If we use the function pointers method as you and Soheil spoke about, a call to arena destruction would do:
- a call to perform gpr_free_aligned() which does a jump to a runtime-initialized function pointer that happens to be aligned_alloc (or whatever alternative) (2 calls)
- on non windows/linux/posix platforms, that alternative would be our in house free_aligned() which does another jump to gpr_free and then another jump to free (4 calls)

Rather, this way it's a single call to a statically known target. I'm not sure what the non-statically intialized global function pointer table actually buys us.

Reply:
Addressed in latest PR.